Sources/PointFreePrelude/Either.swift

```diff
@@ -19,6 +19,15 @@ public func requireSome<A>(_ e: Either<Error, A?>) -> Either<Error, A> {
   }
 }
 
+extension Optional {
+  struct Nil: Error {}
+
+  public func unwrap() throws -> Wrapped {
+    guard let unwrapped = self else { throw Nil() }
+    return unwrapped
+  }
+}
```

Comment:
Helper for unwrapping optionals in throwing contexts. Feels missing from the standard library.